### ISSUE TITLE
Make RouDi Startup Parameters available in TOML Config

### ISSUE THREAD

**Issue by dkroenke** [eclipse/iceoryx#353]
## Brief description
Currently there are parameters passed in different ways to RouDi:
- as command line option in roudi_cmd_line_parser.cpp
- as struct RoudiStartupParameters passed to the roudi ctor in roudi.hpp
- segment/mempool config as C++ API in RouDiConfig available for roudi_main.cpp or as TOML file

We need to wrap up the existing parameters and give the end-user a standardized way to pass them to RouDi. Possible solution would be to use the TOML config for all parameters to have a single point.

- [ ] define where to pass startup parameters to RouDi
- [ ] move struct RoudiStartupParameters to TOML config
- [ ] document for the end-user the possible parameter sets and give example how to use them

**Comment by elBoberido** [comment]
@dkroenke small nitpicking. TOML config is just a serialized RouDiConfig, so that one would be our standardized way. We also need the possibility to set some parameter with the command line, e.g. the log level without constantly editing the the config file
The precedence should be `cmd line parameter > user config file > root config file > compiled into binary`

**Comment by dkroenke** [comment]
@elBoberido Yes you are right, not all parameters should be set in the TOML, the intention is to have clarification where to set which kind of parameters. I agree that things like loglevel or monitoring mode should be set by command line for easier debugging. But there are also things like a unique RouDi ID where the RouDiConfig is maybe the better place.

I have the feeling that in future more parameter options will come up and then it would be good to have clarification where to put them.
